src/main/scala/dibl/Stitches.scala

```diff
@@ -0,0 +1,191 @@
+   * @return
+   */
+  private def splitAssignment(assignment: String): Array[(StitchId, String, String)] = {
+    val (ids, values) = assignment.split("=").partition(_.matches("([a-z]+[0-9]+|cross|twist)"))
+    val (instructions, colors) = values.partition(_.matches("[ctlrp]+"))
+    ids.map(id => (
+      id,
+      instructions.headOption.getOrElse("ctc"),
+      colors.find(availableColors.contains).getOrElse("")
+    ))
+  }
+
+  private def makeValid(s: String, default: String): String = {
+    val strippedPins = s.replaceAll("p", "")
+    if (strippedPins.length == 0) default // need more than a pin
+    else if (s.length - strippedPins.length > 1) default // at most one pin
+    else s
+  }
+
+  /**
+    * @param s a sequence of': (either l's or r's) and/or t's
+    *  @return true if both pairs are twisted at least once
+    */
+  private def bothTwisted(s: String) = s.count('t' == _) > 0
+
+  /**
+    * @param core a sequence of l/r/t bwteen c's
+    * @return true if one of the pair is twisted multiple times
+    */
+  private def repeatedTwisted(core: String) = {
+    core.replaceAll("[cl]","").length > 1 || // mutiple r's and/or t's
+    core.replaceAll("[cr]","").length > 1 // multiple l's and/or t's
+  }
+
+  /** Split  a string (with any sequence of ctlr but at least a c) into
+    * - everything before the first c
+    * - the first c up to and including the last c
+    * - everything after the last c
+    */
+  private val regex: Regex = "([lrt]*)(.*c)?(.*)".r()
+
+  /**
+    * @param stitch Any sequence of "ctrlp" of at least length one.
+    *               Though rl=lr=t the t should be used wherever possible,
+    *               meaning (ignoring pins): between, before and after c's only
+    *               either [l's and/or t's] or [r's and/or t's]
+    * @return An empty string or the default color name for a stitch.
+    *         See also the multicolor system on
+    *         http://susanroberts.info/Working%20diagrams%20-%20part%202.pdf
+    *         archived at
+    *         https://web.archive.org/web/20170916135839/http://susanroberts.info/Working%20diagrams%20-%20part%202.pdf
+    *         We don't have gimps nor tallies. Instead we use brown when
+    *         having repeated twists between two crosses.
+    */
+  def defaultColor(stitch: String) = {
+    val hasPins = stitch.count('p' == _) > 0
+    val crossCount = stitch.count('c' == _)
+    val regex(openTwists, coreStitch, closeTwists) = stitch
+    val twisted = bothTwisted(openTwists) || bothTwisted(closeTwists)
+    (hasPins, crossCount, coreStitch, twisted) match {
+      case (false, 1, _, true) => "green"
+      case (false, 2, "cttc", _) => "turquoise"
+      case (false, 2, core, _) if repeatedTwisted(core) => "brown"
+      case (false, 2, "ctc", true) => "red"
+      case (false, 2, "ctc", false) => "purple"
+      case (false, n, core, _) if n > 2 && core.matches("c(tc)+") => "blue"
+      case _ => ""
```

Comment:
Here the algorithm for the colors. The `regex` on line 179 refers to line 161, The "core" of a stitch starts at the first cross and ends at the last. Difficult to be concise in plain English.